Scripts/DCS-BIOS/lib/modules/aircraft_modules/Ka-50.lua

```diff
@@ -0,0 +1,906 @@
+Ka_50:defineTumb("SHKVAL_OPTICS_ADJ", 8, 3005, 411, 0.1, { 0, 0.7 }, nil, false, "Targeting Display Control Panel", "SHKVAL Optics adjustment selector (no function)")
+--Landing Lights & Voice Warning Control Panel
+Ka_50:defineMultipositionSwitch("LIGHT_LANDING_ON_RETR_OFF", 44, 3001, 382, 3, 0.5, "Landing Lights & Voice Warning Control Panel", "Landing light On/Retraction/Off Switch")
+Ka_50:defineToggleSwitch("LIGHT_LANDING_MAIN_BACKUP", 44, 3002, 383, "Landing Lights & Voice Warning Control Panel", "Main/BackUp landing light Switch")
+Ka_50:defineMultipositionSwitch("ADF_NDB_MODE", 46, 3003, 381, 3, 0.5, "Landing Lights & Voice Warning Control Panel", "ADF Inner-Auto-Outer NDB Switch")
+Ka_50:definePushButton("VOICE_MSG_OFF", 13, 3001, 384, "Landing Lights & Voice Warning Control Panel", "Voice message system messages OFF button")
+Ka_50:defineToggleSwitch("VOICE_MSG_EMER", 13, 3002, 386, "Landing Lights & Voice Warning Control Panel", "Voice message system emergency ON Switch")
+Ka_50:definePushButton("VOICE_MSG_REPEAT", 13, 3003, 385, "Landing Lights & Voice Warning Control Panel", "Voice message system messages repeat button")
+
+----Right Forward Panel
+--EKRAN Warning System display
+---This is an adaptation of parse_indication due to the EKRAN having arrays for some values. In most cases, parse_indication should be used.
+local function parse_EKRAN()
+	local ret = {}
+	local li = list_indication(4)
+	if li == "" then
+		return nil
+	end
+
+	local m = li:gmatch("([^\n]*)\n")
+	local newval = false
+	local name = nil
+	local value = {}
+
+	while true do
+		local line = m()
+		if not line then
+			if name ~= nil then
+				ret[name] = value
+			end
+			break
+		end
+		if line == "-----------------------------------------" then
+			newval = true
+			if name ~= nil then
+				ret[name] = value
+				name = nil
+				value = {}
+			end
+		else
+			if newval == true then
+				newval = false
+				name = line
+			else
+				value[#value + 1] = line
+			end
+		end
+	end
+	return ret
+end
+local indEKRAN = nil
+moduleBeingDefined.exportHooks[#moduleBeingDefined.exportHooks + 1] = function()
+	indEKRAN = parse_EKRAN()
+end
+local function getEKRAN_memory()
+	if indEKRAN == nil or indEKRAN.txt_memory == nil then
+		return "0"
+	end
+	return "1"
+end
+local function getEKRAN_queue()
+	if indEKRAN == nil or indEKRAN.txt_queue == nil then
+		return "0"
+	end
+	return "1"
+end
+local function getEKRAN_failure()
+	if indEKRAN == nil or indEKRAN.txt_failure == nil then
+		return "0"
+	end
+	return "1"
+end
+Ka_50:defineString("EKRAN_MEMORY", getEKRAN_memory, 1, "EKRAN", "Memory message")
+Ka_50:defineString("EKRAN_QUEUE", getEKRAN_queue, 1, "EKRAN", "Queue message")
+Ka_50:defineString("EKRAN_FAILURE", getEKRAN_failure, 1, "EKRAN", "Failure message")
+local function getEKRAN_txt1_line1()
+	if indEKRAN == nil or indEKRAN.txt_1 == nil then
+		return "          "
```

Comment:
I was going to make this suggestion everywhere, but there are simply too many places. Basically anywhere we return these blank strings of arbitrary length, we can just return `""`. The output of `defineString` handles padding the string to the required length automatically.
```suggestion
		return ""
```